frontend/ui/widget/checkbutton.lua

```diff
@@ -56,8 +48,6 @@ function CheckButton:initCheckButton(checked)
         checked = self.checked,
         enabled = self.enabled,
         face = self.face,
-        parent = self.parent or self,
-        show_parent = self.show_parent or self,
```

Comment:
Will need a little attention from @NiLuJe : there are mentions of show_parent in UIManager.lua - so, even if unused here, it's possible some upper widget may go down to expect one of these.
I don't think it's the case, but better be sure.

Reply:
I've tested almost all places with checkbuttons (there are 6, can reach 5, except warmlight `Auto` button).

Reply:
In general, should we always pass `parent = ` to the checkbutton?
Currently we have 3 calls with it and 3 without.

Reply:
Yes, we want to have that available as a standard thing.

Reply:
Old modules don't have it: Open with... dialog, Show password (inputtext), Frontlight widget.

Reply:
Well, I'm thinking it's easier to just have it but maybe there are differing opinions?

Reply:
> Old modules don't have it

Guess it depends on where these were copied from when made :)
Usually, the parent stuff is there so a widget can reference its parent for setDirty/refresh so the full parent region gets refreshed/repainted.
So, if it's not used in the module, I guess they could be removed - unless (my question to @NiLuJe) some other stuff may expect them to be there in any widget, or do something better if they are there.

So, while we're in doubt, best to keep them :)

Reply:
I mean mainly that then while coding you can refer to parent without thinking instead of going through a (however brief) "why doesn't this work" spiel. It's happened to me before.

Reply:
Regardless whether or not it's actually used anywhere *right now*.

That's (mostly) different from some of the other things in here like padding.

Reply:
So, ready.

Reply:
Yeah, I'd rather have it in, especially for widgets that serve an actual UI purpose (as opposed to, say, containers), even if it doesn't appear to be used right now ;).

(Especially here, because this is useful when cross-referencing with other *Button widgets ;)).